### PR TITLE
Init tests for `mongodump` and `mongorestore` utility

### LINKED ISSUE #149: Add tests for `mongorestore` and `mongodump`
We should run `mongorestore` and `mongodump` with various flags to ensure we stay compatible with FerretDB. Let's add a new set of tests that would run binaries, not use their tests.

* https://www.mongodb.com/docs/database-tools/mongorestore/
* https://www.mongodb.com/docs/database-tools/mongodump/
* https://github.com/mongodb/mongo-tools
* https://github.com/mcampo2/mongodb-sample-databases

### PR DESCRIPTION
**Pull request by noisersup**
Closes #149.

### CODE REVIEW

**Comment by AlekSi** [comment]
@noisersup those are not diff tests, that should a separate directory

**Review by AlekSi — COMMENTED** [review]
```
    extra_hosts:
      - "host.docker.internal:host-gateway"
```

And let's maybe switch from old `docker-compose` to new `docker compose`

**Review by AlekSi — COMMENTED** [review]
Please remove dependency on github.com/docker/docker package. `docker-compose exec` or `docker compose exec` should be executed using `os.Exec`

**Comment by noisersup** [comment]
@AlekSi Based on your changes, is running `mongodump` on `mongosh` container is still relevant idea?

**Comment by AlekSi** [comment]
We already have a container with those tools, we don't need to create another one

**Comment by noisersup** [comment]
Which container?

**Comment by noisersup** [comment]
Current state: Locally on a main branch I get the following error:
> Failed: error dumping metadata: error getting indexes for collection `test.values`: invalid DocumentSequence: a non-document value was found in sequence

Though, I cannot replicate it on dance tests

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,7 +1,6 @@
 # Contributing
 
 The dance tool and tests run on the host; macOS, Linux, and Windows are expected to work.
-Databases under test (FerretDB and MongoDB) may be running on the host or inside Docker; Docker Compose configuration is provided for convenience but not required.
```
Comment (noisersup) [inline CONTRIBUTING.md:4]: > Docker Compose configuration is provided for convenience but not required.
It's required now to run tests/mongo-tools

**Review by w84thesun — COMMENTED** [review]
That's a lot of work! Asked about CONTRIBUTING.md change.

File: CONTRIBUTING.md (hunk 1 above)
Comment (noisersup) [reply CONTRIBUTING.md:4]: Yeah, let's mention it, though it's still temporary and should be replaced in another PR

**Review by w84thesun — DISMISSED** [review]

**Review by rumyantseva — DISMISSED** [review]
LGTM.

The current approach works for me locally when I have FerretDB running on my host (port 27017):

- For `diff` tests everything works as before - no `env-up` in dance needed etc 
- For `mongo-tools`: 
    - From the `dance` directory I run `docker compose up mongosh`
    - Then I run `task dance DB=ferretdb TEST=mongo-tools` and it passes!
    - In the logs of my local FerretDB instance I see that it was called and reached.

File: Taskfile.yaml [hunk 2]
```diff
@@ -27,6 +27,11 @@ tasks:
     cmds:
       - docker compose logs --follow
 
+  env-mongosh:
+    desc: "Start mongosh container"
+    cmds:
+      - docker compose up mongosh -d
+
```
Comment (noisersup) [inline Taskfile.yaml:34]: Okay, so the solution to all of local instance issues is just a new task (which is now in deps of `task dance`). Thanks @rumyantseva for reminding that it's possible to run a single container from docker-compose :slightly_smiling_face:

Comment (rumyantseva) [reply Taskfile.yaml]: Yeah, the next step would be to not run it at all (as @AlekSi mentioned on Slack).

Comment (noisersup) [reply Taskfile.yaml]: done

**Review by AlekSi — APPROVED** [review]
Let's start with that

**Review by w84thesun — APPROVED** [review]

**Pull request merged**